File: server/schemas/enrichment.py
"""Pydantic schemas for enrichment API endpoints."""

from pydantic import BaseModel, Field


class EnrichmentRequest(BaseModel):
    """Request to enrich a product with specifications."""

    manufacturer: str = Field(default="", description="Product manufacturer")
    model_number: str = Field(..., description="Product model/part number")
    product_name: str = Field(default="", description="Optional product name hint")


class EnrichmentResponse(BaseModel):
    """Response with enriched product data."""

    enriched: bool = Field(description="Whether enrichment was successful")
    source: str = Field(description="Source of enrichment data (ai_knowledge, cache, none)")
    name: str = Field(description="Full product name")
    description: str = Field(default="", description="Product description")
    features: list[str] = Field(default_factory=list, description="Product features")
    msrp: float | None = Field(default=None, description="Original MSRP if known")
    release_year: int | None = Field(default=None, description="Release year if known")
    category: str = Field(default="", description="Product category")
    confidence: float = Field(default=0.0, description="Confidence score 0-1")
    formatted_description: str = Field(
        default="", description="Pre-formatted description for Homebox"
    )


class EnrichmentSettingsRequest(BaseModel):
    """Request to update enrichment settings."""

    enabled: bool = Field(description="Whether enrichment is enabled")
    auto_enrich: bool = Field(
        default=False, description="Automatically enrich items after detection"
    )
    cache_ttl_hours: int = Field(
        default=24, ge=1, le=168, description="Cache time-to-live in hours (1-168)"
    )


class EnrichmentSettingsResponse(BaseModel):
    """Response with current enrichment settings."""

    enabled: bool = Field(description="Whether enrichment is enabled")
    auto_enrich: bool = Field(description="Automatically enrich items after detection")
    cache_ttl_hours: int = Field(description="Cache time-to-live in hours")
    cache_entries: int = Field(description="Number of cached enrichment results")


class ClearCacheResponse(BaseModel):
    """Response from clearing the enrichment cache."""

    cleared_count: int = Field(description="Number of cache entries cleared")
    message: str = Field(description="Status message")

File: frontend/src/lib/api/enrichment.ts
/**
 * Enrichment API endpoints
 *
 * Provides AI-powered product specification lookup using the configured AI provider.
 * Uses the AI's training knowledge to find product details - no external APIs required.
 */

import { request } from './client';

// =============================================================================
// TYPES
// =============================================================================

export interface EnrichmentRequest {
	manufacturer: string;
	model_number: string;
	product_name?: string;
}

export interface EnrichmentResponse {
	/** Whether enrichment found useful data */
	enriched: boolean;
	/** Source of data: 'ai_knowledge', 'cache', or 'none' */
	source: string;
	/** Full product name */
	name: string;
	/** Product description */
	description: string;
	/** Product features list */
	features: string[];
	/** Original MSRP if known */
	msrp: number | null;
	/** Release year if known */
	release_year: number | null;
	/** Product category */
	category: string;
	/** Confidence score 0-1 */
	confidence: number;
	/** Pre-formatted description for Homebox (combines description + features) */
	formatted_description: string;
}

export interface ClearCacheResponse {
	cleared_count: number;
	message: string;
}

// =============================================================================
// API FUNCTIONS
// =============================================================================

/**
 * Enrich a product with detailed specifications.
 *
 * Uses the configured AI provider to look up product details from its training data.
 * Results are cached locally to avoid repeated API calls.
 *
 * @param data - Product identification (manufacturer, model number, optional name)
 * @returns Enriched product data or empty result if not found
 */
export const enrichProduct = (data: EnrichmentRequest) =>
	request<EnrichmentResponse>('/enrichment/lookup', {
		method: 'POST',
		body: JSON.stringify(data),
	});

/**
 * Clear the enrichment cache.
 *
 * @returns Count of cleared entries
 */
export const clearEnrichmentCache = () =>
	request<ClearCacheResponse>('/enrichment/cache', {
		method: 'DELETE',
	});
